scanner/decisions/0009-scannerv4-read-manual-vulnerabilities.md

```diff
@@ -0,0 +1,69 @@
+# 0009 - Improve Scanner V4 manually added vulnerabilities updating process
+
+- **Author(s):** Yi Li <yli3@redhat.com>
+- **Created:** [2024-04-30]
+
+## Context
+
+Currently, Scanner v4 includes a feature for manually updating vulnerabilities to add and update urgent vulnerability data. 
+This allows clients to access the most recent vulnerability information before it becomes officially available in some data sources. 
+However, this manual update process has a major flaw: the vulnerability data is integrated with the codebase, which means clients cannot access the updated data unless they install a patch release that includes these vulnerabilities. 
```

Comment:
I meant that users/customers don't have to install a new patch release that includes the manual vulnerabilities. They need a patch release so that they have access to the latest vulnerability updates that contain the new vulnerabilities. The problem description is misleading.

Reply:
Sorry, I am not sure which part is misleading. 

> Clients cannot access the updated data unless they install a patch release that includes these vulnerabilities
Clients don't have to upgrade to the new patch release (unless they want the newly added vulnerabilities. )

Reply:
> Sorry, I am not sure which part is misleading.

The part that says "a patch release that includes these vulnerabilities". The patch release itself does not contain the manual vulnerabilities. These are served in the vulnerability bundle, which is pulled from the service. But, they have do have to upgrade to a new patch release to be able to consume the new bundle that contains those vulns, because we use versions to select which bundle to pull.

Reply:
> The patch release itself does not contain the manual vulnerabilities.
But isn't `vulns.go` integrated in the code, which is part of the patch release?

Reply:
Yes, but we consume the data from the vulnerability bundle, not from `vulns.go`, right?

Reply:
Example: If we could merge commits to a patch release branch (we don't have that concept today) and tag it with something greater than `4.4.1` (e.g., `4.4.1.v`), it would be possible for the vulnerability generation workflow to pull that tag instead of `4.4.1`, and we would not need to force customers to upgrade because the vulnerability bundle would be updated for the `4.4.1` bundle. That is complicated and error-prone, and I prefer your proposal. But do you see how your statement is a delightfully ambiguous :sweat_smile:?

Reply:
> which means clients cannot access the updated data unless they install a patch release that fetches these vulnerabilities

Is this more reasonable?

Reply:
```suggestion
However, this manual update process has a significant flaw. The vulnerability data is integrated with the codebase, which means that updating manual vulnerabilities requires a code change. Currently, the vulnerability bundle workflow serves one vulnerability bundle per X.Y.Z stream and uses release tags to generate the bundle. Thus, changing each release branch, publishing a new patch release, and upgrading clients to start using the new bundle version would be necessary. This approach restricts clients' ability to access the new bundle until they upgrade.```

Disclaimer: I generated this with a public AI service. But it communicates the actual issue, and clarifies better what problem this ADR is trying to solve.